packages/core/src/RenderingEngine/BaseVolumeViewport.ts

```diff
@@ -264,20 +264,40 @@ abstract class BaseVolumeViewport extends Viewport implements IVolumeViewport {
     cfun.applyColorMap(colormapObj);
     cfun.setMappingRange(range[0], range[1]);
     volumeActor.getProperty().setRGBTransferFunction(0, cfun);
+  }
 
-    const ofun = vtkPiecewiseFunction.newInstance();
-    ofun.addPoint(range[0], 0.0);
-    ofun.addPoint(range[1], 1.0);
-    volumeActor.getProperty().setScalarOpacity(0, ofun);
-
-    if (!opacityMapping) {
+  /**
+   * Sets the opacity for the volume with the given ID.
+   *
+   * @param colormap -  An object containing opacity or opacityMapping properties.
+   * @param volumeId - The ID of the volume to set the opacity for for.
+   *
+   * @returns void
+   */
+  private setOpacity(colormap: ColormapPublic, volumeId: string) {
```

Comment:
You are using only .opacity from the colormap, so it should just be that in the arguments too

`setOpacity(opacity: number | OpacityMapping [], volumeId?: string)`